edb/common/ordered.py

```diff
@@ -28,35 +28,29 @@
 
 
 class OrderedSet(MutableSet[K]):
+
+    map: Dict[K, None]
+
     def __init__(self, iterable: Optional[Iterable[K]] = None) -> None:
-        self.map: collections.OrderedDict[K, K] = collections.OrderedDict()
         if iterable is not None:
-            # The ignore below is because typing of collections.abc.MutableSet
-            # inherits a limitation of the built-in set that disallows |= with
-            # iterables that are not sets themselves.  However, the mixin
-            # *does* allow this and OrderedSet depends on this.
-            self.update(iterable)
+            self.map = {v: None for v in iterable}
+        else:
+            self.map = {}
 
-    def add(self, item: K, *, last: Optional[bool] = None) -> None:
-        self.map[item] = item
-        if last is not None:
-            self.map.move_to_end(item, last=last)
+    def add(self, item: K) -> None:
+        self.map[item] = None
```

Comment:
You can push an item to be first easily:

```
>>> a = {'a': 1, 'b': 2, 'c': 3}
>>> b = {'c': None}
>>> b.update(a)
>>> b
{'c': 3, 'a': 1, 'b': 2}
```

You can also push it last, if you just reconstruct the dict. Both are O(n) technically, although the code sample above should be fast enough as it's all in C.

Reply:
Yes, but we don't need this anywhere.  The only place where this is used is DDL subcommands, and there I just used a list.